test/ex_machina/ecto_test.exs

```diff
@@ -90,9 +131,10 @@ defmodule ExMachina.EctoTest do
 
     user = ExMachina.Ecto.assoc(MyApp.EctoFactories, attrs, :user)
 
-    created_user = %{id: 3, name: "John Doe", admin: false}
+    vals = %{name: "John Doe", admin: false}
+    created_user = struct(MyApp.User, vals) |> Map.put(:id, 1)
     assert user == created_user
-    assert_received {:created, ^created_user}
+    assert_received {:created, %MyApp.User{}}
```

Comment:
I think you could match against the created_user `{:created, ^created_user}` in this test and the one below